rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,249 @@
+				+ `${additionalPolyfillPatterns[feature] || ''})$`,
+			'i',
+		),
+	};
+});
+
+function report(context, node, feature) {
+	let [, namespace, method = ''] = feature.split('.');
+	if (namespace === 'typed-array' && method.endsWith('-array')) {
+		namespace = method;
+		method = '';
+	}
+
+	context.report({
+		node,
+		messageId: MESSAGE_ID_POLYFILL,
+		data: {
+			featureName: `${constructorCase(namespace)}${method && '#'}${camelCase(method)}`,
+		},
+	});
+}
+
+class SemverNodeVersion {
+	constructor(nodeVersion) {
+		this.nodeVersion = nodeVersion;
+		this.validNodeVersion = semver.coerce(nodeVersion);
+	}
+
+	compare(featureVersion) {
+		const supportedNodeVersion = semver.coerce(featureVersion);
+		return this.validNodeVersion
+			? semver.lte(supportedNodeVersion, this.validNodeVersion)
+			: semver.ltr(supportedNodeVersion, this.nodeVersion);
+	}
+}
+
+function processRule(context, node, moduleName, targets) {
+	if (!moduleName || typeof moduleName !== 'string') {
+		return;
+	}
+
+	const nodeVersion = targets instanceof SemverNodeVersion && targets;
+	const importedModule = moduleName.replace(/([/\\].+?)\.[^.]+$/, '$1');
+
+	const unavailableFeatures = coreJsCompat({targets}).list;
+	const coreJsModuleFeatures = coreJsEntries[importedModule.replace('core-js-pure', 'core-js')];
+
+	const checkFeatures = features => nodeVersion
+		? features.every(feature => compatData[feature].node && nodeVersion.compare(compatData[feature].node))
+		: !features.every(feature => unavailableFeatures.includes(feature));
+
+	if (coreJsModuleFeatures) {
+		if (coreJsModuleFeatures.length > 1) {
+			if (checkFeatures(coreJsModuleFeatures)) {
+				context.report({
+					node,
+					messageId: MESSAGE_ID_CORE_JS,
+					data: {
+						coreJsModule: moduleName,
+					},
+				});
+			}
+		} else if (nodeVersion ? nodeVersion.compare(compatData[coreJsModuleFeatures[0]].node) : !unavailableFeatures.includes(coreJsModuleFeatures[0])) {
+			report(context, node, coreJsModuleFeatures[0]);
+		}
+
+		return;
+	}
+
+	const polyfill = polyfills.find(({pattern}) => pattern.test(importedModule));
+	if (polyfill) {
+		const [, namespace, method = ''] = polyfill.feature.split('.');
+		const [, features] = Object.entries(coreJsEntries).find(it => it[0] === `core-js/actual/${namespace}${method && '/'}${method}`);
+		if (checkFeatures(features)) {
+			report(context, node, polyfill.feature);
+		}
+	}
+}
+
+function create(context) {
+	const options = context.options[0];
+	let targets = options && options.targets;
+	if (!targets) {
+		targets = getTargetsFromPackage(context.getFilename(), options && options.useBrowserlistFieldByDefault);
+	} else if (options.treatsTargetsAsSemver) {
+		targets = new SemverNodeVersion(targets);
+	}
+
+	if (!targets) {
+		return {};
+	}
+
+	return {
+		'CallExpression[callee.name="require"]'(node) {
+			processRule(context, node, node.arguments[0].value, targets);
```

Comment:
```suggestion
		[STATIC_REQUIRE_SOURCE_SELECTOR](node) {
```

`require()` arguments can be anything.